ext/zlib/zlib_filter.c

```diff
@@ -90,6 +90,7 @@ static php_stream_filter_status_t php_zlib_inflate_filter(
 				exit_status = PSFS_PASS_ON;
 			} else if (status != Z_OK) {
 				/* Something bad happened */
+				php_error_docref(NULL, E_NOTICE, "zlib inflate failed");
```

Comment:
Looks like zlib does have a `zError()` function we can use here.

Reply:
Ah, thanks! Not quite as useful as I'd had hoped, though ("data error").